array.go

```diff
@@ -90,12 +91,12 @@ func (a *Array) RawJSON(val []byte) *Array {
 // Err serializes and appends the err to the array.
 func (a *Array) Err(err error) *Array {
 	switch m := ErrorMarshalFunc(err).(type) {
+	case nil:
+		a.buf = enc.AppendNil(enc.AppendArrayDelim(a.buf))
 	case LogObjectMarshaler:
 		a = a.Object(m)
 	case error:
-		if m == nil || isNilValue(m) {
-			a.buf = enc.AppendNil(enc.AppendArrayDelim(a.buf))
```

Comment:
Nothing I could do in testing would ever hit this line. It simply doesn't ever not match the `case nil:`

Reply:
> Nothing I could do in testing would ever hit this line. It simply doesn't ever not match the `case nil:`

Now have **all** of these tested _except_ for the `Context` version. Still can't seem to hit that one, but `Event`, `Fields`, and `Array` are now covered.